### PR TITLE
feat: allow named ssrcs in TrackInfo plain objects

### PR DESCRIPTION
**Pull request by murillo128**

### CODE REVIEW

File: lib/TrackInfo.js [hunk 1]
```diff
@@ -257,10 +257,44 @@ TrackInfo.expand = function(plain)
...
+	} else {
+		//Check if we have named ssrcs
+		if ("media" in plain.ssrcs) 
+		{
+			//Get ssrc for media
+			const media = plain.ssrcs.media;
+			//Append to ssrc list
+			trackInfo.addSSRC(media);
+			//Check if we have named rtx ssrc
+			if ("rtx" in plain.ssrcs)
```
Comment (mildsunrise) [inline lib/TrackInfo.js]: plays better with the type system :innocent: 
```suggestion
			if (plain.ssrcs.rtx !== undefined) 
```

File: lib/TrackInfo.js [hunk 2]
```diff
@@ -257,10 +257,44 @@ TrackInfo.expand = function(plain)
...
+			{
+				//Get ssrc for rtx
+				const rtx = plain.ssrcs.rtx;
+				//Append to ssrc list
+				trackInfo.addSSRC(rtx);
+				//Create a FID group
+				trackInfo.addSourceGroup(new SourceGroupInfo("FID",[media,rtx]));
+			}
+			//Check if we have named fec ssrc
+			if ("fec" in plain.ssrcs)
```
Comment (mildsunrise) [inline lib/TrackInfo.js]:
```suggestion
			if (plain.ssrcs.fec !== undefined) 
```

File: lib/TrackInfo.js [hunk 3]
```diff
@@ -257,10 +257,44 @@ TrackInfo.expand = function(plain)
...
+	//If we have an array of ssrcs
+	if (Array.isArray(plain.ssrcs))
+	{
+		//for each ssrc
+		for (const ssrc of plain.ssrcs)
+			//Add ssrc
+			trackInfo.addSSRC(ssrc);
+	} else {
+		//Check if we have named ssrcs
+		if ("media" in plain.ssrcs)
```
Comment (mildsunrise) [inline lib/TrackInfo.js]:
```suggestion
		if (plain.ssrcs && plain.ssrcs.media !== undefined) 
```

File: lib/TrackInfo.js [hunk 4]
```diff
@@ -257,10 +257,44 @@ TrackInfo.expand = function(plain)
...
-		trackInfo.addSSRC(ssrc);
+
+	//If we have an array of ssrcs
+	if (Array.isArray(plain.ssrcs))
+	{
+		//for each ssrc
+		for (const ssrc of plain.ssrcs)
+			//Add ssrc
+			trackInfo.addSSRC(ssrc);
+	} else {
```
Comment (mildsunrise) [inline lib/TrackInfo.js]: maybe convert the

``` js
} else {
  if (...) {
    // ...
  }
}
```

into

``` js
} else if (...) {
  // ...
}
```

for readability?

**Review by mildsunrise — APPROVED** [review]
lgtm, only minor type/syntax related changes

File: lib/TrackInfo.js (hunk 1 above)
Comment (murillo128) [reply lib/TrackInfo.js:277]: rtx "could" be 0

**Comment by murillo128** [comment]
thanxs!

**Pull request merged**